project-koku/koku
# COST-2632 AWS Org Unit and Account filter are not creating an additive set for the API query

## Jira Ticket

[COST-2632](https://issues.redhat.com/browse/COST-2632)

## Description

This change will ...
- add a function to get accounts in an org unit tree to fix access list when filtering on that org unit and grouping by an account, for a user with restricted access.

## Testing

1. Checkout Branch
2. Restart Koku
3. Update DEFAULT_IDENTIY in settings with  an org unit and an account outside that org unit, as the values below
```
"is_org_admin": "False", 
"access": {
        "aws.account": {"read": ["9999999999993"]}, 
        "aws.organizational_unit": {"read": ["OU_001"]}
   }
```
4. Hit endpoint (replace org_unit_id and account parameter with appropriate values)

>   127.0.0.1:8000/api/cost-management/v1/reports/aws/costs/?filter[resolution]=monthly&filter[time_scope_units]=month&filter[time_scope_value]=-1&cost_type=unblended_cost&filter[org_unit_id]=OU_001&group_by[account]=9999999999993


    1. You should see total cost from both the org unit and the accounts outside its tree.
    
5. Update the settings in step 3 to the default admin permissions, `"is_org_admin": "True", "access": {}`, and hit the endpoint in step 4. The response should be the same.

## Notes

...
